src/tests/system/tests/test_passkey.py

```diff
@@ -0,0 +1,480 @@
+        4. Add 'pam_passkey_auth = True' under pam section of sssd.conf file.
+    :steps:
+        1. Check the umockdev-run command for authentication of the user.
+        2. Check umockdev-run that the output contains the username.
+        3. Make server not reachable by removing port of LDAP(389) and LDAP(636) from server.
+        4. Check authentication by running umockde-run command.
+    :expectedresults:
+        1. Command returns 0.
+        2. Output contains username of the user.
+        3. Successfully made server offline, by removing port using firewalld commands.
+        4. umockdev-run command is successful and returning username.
+    :customerscenario: False
+    """
+    tc = passkey_str = username = ""
+    if isinstance(provider, IPA):
+        tc = "su_ipa_test3"
+        username = 'user1@ipa.test'
+        passkey_str = "passkey:NUZMRUXIb/W8Ij1GqwCDHSCWxt/SxWxckwtQjLYi/X6Y1qZFB+HI8WO6khzAjzsz248" \
+                      "kHbaeAf9qfmqfCky1Jg==,MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEIasAa8ogjPCKXeA4K" \
+                      "Y3t0W3xBRmG+E4D+MNoRIAJrYuNLSYtAcOL7DCbIfgc+7c5Y4Mh/FzoEyeumKGYMoyTfg=="
+
+    elif isinstance(provider, LDAP):
+        tc = "su_ldap_test6"
+        username = "user1@test"
+        passkey_str = "passkey:mQEUTWdtDJPELQNTDdxXNHlfIO1qXFf0LVZjWEfyDALFzvLZ4e4XD5bemqq+o3ThrzT6k1I1n3Z2N00G" \
+                      "vLSmjQ==,MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEqk7K5VAI7Evr4ar8X82L/sxm/Bnm5Ti31xnLfGO0BipwH" \
+                      "ucw8+/wT4+6T9j5gdMwZKUcXR4BILpmULEyrcZUfw=="
+
+    user = provider.user('user1').add()
+    user.passkey_add(passkey_str)
+
+    client.sssd.pam['pam_passkey_auth'] = 'true'
+    client.sssd.start()
+
+    for record_file in [f'{tc}.ioctl', f'{tc}.script', f'{tc}.device']:
+        src_file = f"{os.getcwd()}/tests/data/passkey/{tc}/{record_file}"
+        client.fs.upload(src_file, f"/tmp/{record_file}")
+
+    output1 = client.auth.passkey.su(tc, username, 123456)
+
+    # Added firewalld command to remove ldap and ldaps port.
+
+    for cmd in ['dnf install firewalld -y', 'systemctl start firewalld',
+                'firewall-cmd --permanent --remove-port=636/tcp',
+                'firewall-cmd --permanent --remove-port=389/tcp', 'firewall-cmd --reload']:
+        provider.host.ssh.run(cmd, raise_on_error=False)
+
+    output2 = client.auth.passkey.su(tc, username, 123456)
+
+    for cmd in ['firewall-cmd --permanent --add-port=636/tcp',
+                'firewall-cmd --permanent --add-port=389/tcp',
+                'firewall-cmd --reload', 'systemctl stop firewalld']:
+        provider.host.ssh.run(cmd, raise_on_error=False)
+
+    assert output1.stdout_lines[-1] == output2.stdout_lines[-1] == 'user1'
+
+
+@pytest.mark.tier(1)
+@pytest.mark.topology(KnownTopology.IPA)
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_passkey__su_fail_check_when_srv_offline(client: Client, provider: GenericProvider):
```

Comment:
same as above.  rename to something like test_passkey__su_fail_check_when_srv_not_resolvable